src/CMakeLists.txt

```diff
@@ -8,6 +8,24 @@ add_library(VulkanMemoryAllocator
     ${PROJECT_SOURCE_DIR}/include/vk_mem_alloc.h
 )
 
+option(VMA_BUILD_WERROR "Treat compiler warnings as errors")
+if (VMA_BUILD_WERROR)
+    target_compile_options(VulkanMemoryAllocator PRIVATE
+        "$<IF:$<CXX_COMPILER_ID:MSVC>,/WX,-Werror>"
+    )
+endif()
+
+if(${CMAKE_CXX_COMPILER_ID} MATCHES "(GNU|Clang)")
+    target_compile_options(VulkanMemoryAllocator PRIVATE
+        -Wpedantic
+        -Wall
+        -Wextra
+        -Wno-unused-parameter
+        -Wno-unused-variable
+        -Wno-missing-field-initializers
```

Comment:
My goal is to fix the remaining warnings in another PR. I don't want to complicate this one too much.